lib/emque/producing/message/message.rb

```diff
@@ -97,21 +97,26 @@ def to_json
         Oj.dump(data, :mode => :compat)
       end
 
-      def publish(publisher=Emque::Producing.publisher)
+      def publish(publisher=nil)
+        publisher ||= Emque::Producing.publisher
         log "publishing...", true
         if valid?
           log "valid...", true
           if Emque::Producing.configuration.publish_messages
             sent = publisher.publish(topic, message_type, to_json, partition_key)
             log "sent #{sent}"
-            if raise_on_failure? && !sent
-              raise MessagesNotSentError.new
-            end
+            raise MessagesNotSentError.new if raise_on_failure? && !sent
           end
         else
           log "failed...", true
           raise InvalidMessageError.new(invalid_message)
```

Comment:
If the message has raise_on_failure = false and it's invalid then it will raise that error. It should be
```ruby
raise InvalidMessageError.new(invalid_message) if raise_on_failure?
```